xdsl/dialects/cf.py

```diff
@@ -91,12 +110,218 @@ def __init__(
     """
 
 
+@irdl_op_definition
+class Switch(IRDLOperation):
+    """Switch operation"""
+
+    name = "cf.switch"
+
+    case_values = opt_attr_def(DenseIntOrFPElementsAttr)
+
+    flag = operand_def(AnySignlessIntegerOrIndexType)
+
+    default_operands = var_operand_def(AnyAttr())
+
+    case_operands = var_operand_def(AnyAttr())
+
+    # Copied from AttrSizedSegments
+    case_operand_segments = attr_def(attr_constr_coercion(DenseArrayBase))
+
+    default_block = successor_def()
+
+    case_blocks = var_successor_def()
+
+    irdl_options = [AttrSizedOperandSegments()]
+
+    traits = frozenset([IsTerminator()])
+
+    def __init__(
+        self,
+        flag: Operation | SSAValue,
+        default_block: Successor,
+        default_operands: Sequence[Operation | SSAValue],
+        case_values: DenseIntOrFPElementsAttr | None = None,
+        case_blocks: Sequence[Successor] = [],
+        case_operands: Sequence[Sequence[Operation | SSAValue]] = [],
+        attr_dict: dict[str, Attribute] | None = None,
+    ):
+        case_operand_segments: list[int] = []
+        c_operands: list[Operation | SSAValue] = []
+        for case_operand in case_operands:
+            case_operand_segments.append(len(case_operand))
+            c_operands += list(case_operand)
+        operands = [flag, default_operands, c_operands]
+        attributes: dict[str, Attribute] = attr_dict or {}
```

Comment:
Instead of always creating a new dict if the one passed in is empty, only create one if the one passed in is None
```suggestion
        attributes = attr_dict if attr_dict is not None else
```

Reply:
I don't understand this, python's `or` should be lazy

Reply:
```
>>> x = {}
>>> y = x or {}
>>> x is y
False
```

Reply:
I see, thanks